hs628_tools/normalization_tool.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import sklearn as sk
+from sklearn import preprocessing
+
+
+class NormCheck:
+    def __init__(self, alldata, cutoff=0.3, alldata1=None):
+        self.alldata = alldata
+        self.cutoff = cutoff
+        self.alldata1 = alldata1
+
+    def find_percent_diff(self):
+        """ 
+        This function takes the data as a numpy.ndarray. It calculates mean for each column.
+        It further processes this data to return the percentage difference.
+        """
+        rowwise_missing_entries = len(alldata) - np.sum(alldata == alldata, axis=0)
+        data_nocategory_columns = rowwise_missing_entries < len(alldata)
+        self.alldata1 = alldata[:, data_nocategory_columns]
+        allmean = self.alldata1.mean(axis=0)
+        allmeanmin = allmean.min()
+        allmeanmax = allmean.max()
+        percent_diff = allmeanmin * 100 / float(allmeanmax)
+        return percent_diff
+
+    def get_input(self, percent_diff):
+        """
+        This function presents user to provide an input, if it determines that data will benefit from normalization.
+        The user input gives choice to user if they want to proceed with normalization or not.
+        If data does not need normalization, the function returns with an informative message reflecting the same.
+        """
+        if percent_diff < self.cutoff:
+            rescale = raw_input ("This dataset should be normalized. Do you want to proceed with normalization?"+"\n"+"YES/NO"+"\n")
+            return rescale
+        else:
+            return "This dataset may not need normalization."
+
+    def compare(self, rescale):
```

Comment:
Since I changed the flow a couple of times, the names definitely don't do proper justice to what the function is completely doing. This function returns normalized data to the user. I will rename it so that it is clear.